ansible.go

```diff
@@ -4,65 +4,105 @@ package main
 
 import (
 	"encoding/json"
-	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
-	"github.com/spf13/viper"
+	"net"
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
 )
 
 // AnsibleConfig is a collection of meta-information about an Ansible repository.
 //
 // The virtualenv specified by the VenvConfig needs to be initialized before running Ansible commands.
 // All dir paths are relative to the root of the source tarball.
 type AnsibleConfig struct {
-	VenvConfig         VenvConfig // Virtualenv config that Ansible will be executed in
-	Cwd                string     // Path to change to when running Ansible commands
-	InventoryList      []string   // Paths to all desired inventories
+	VenvConfig    VenvConfig // Virtualenv config that Ansible will be executed in
+	Cwd           string     // Path to change to when running Ansible commands
+	InventoryList []string   // Paths to all desired inventories
+}
+
+// CreateAnsibleTargetsList generates and returns an array of possible targets
+// The possible targets are ip addresse from the host interfaces or the hostname itself
+func CreateAnsibleTargetsList() []string {
+	var inventoryTargets = []string{}
+
+	// extract all ip addresses except the loopback one and add the hostname as a last fallback mechanism
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		logrus.Fatal("Unable to get the network interfaces")
+	}
+	for _, i := range ifaces {
+		addrs, err := i.Addrs()
+		if err != nil {
+			logrus.Fatal("Unable to extract the ip addresses")
+		}
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
+			if ip == nil || ip.IsLoopback() {
+				continue
+			}
+			inventoryTargets = append(inventoryTargets, ip.String())
+		}
+	}
+	inventoryTargets = append(inventoryTargets, hostname)
+	return inventoryTargets
 }
 
-// FindInventoryForHost takes a hostname and returns the inventory where the host was found.
+// FindInventoryForHost calls CreateAnsibleTargetsList to get a list of possible targets
+// and returns the traget and inventory where the target was found.
```

Comment:
nit: typo
```suggestion
// and returns the target and inventory where the target was found.
```

Reply:
Also, this line makes it seem like we're returning the `target, inventory`, but we're returning `inventory, target`. Should probably fix the ordering to avoid confusion.

Also a little more description would be nice.
"returns the path to the inventory and the matched node identifier for this host" or whatever would be accurate.